src/Repository/EventDispatcherAwareFeatureRepository.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Pheature\Crud\Toggle\Repository;
+
+use Pheature\Core\Toggle\Write\Feature;
+use Pheature\Core\Toggle\Write\FeatureId;
+use Pheature\Core\Toggle\Write\FeatureRepository;
+use Psr\EventDispatcher\EventDispatcherInterface;
+
+final class EventDispatcherAwareFeatureRepository implements FeatureRepository
+{
+    private FeatureRepository $featureRepository;
+    private EventDispatcherInterface $eventDispatcher;
+
+    public function __construct(FeatureRepository $featureRepository, EventDispatcherInterface $eventDispatcher)
+    {
+        $this->featureRepository = $featureRepository;
+        $this->eventDispatcher = $eventDispatcher;
+    }
+
+    public function save(Feature $feature): void
+    {
+        $this->featureRepository->save($feature);
+
+        $this->dispatchEvents($feature->release());
+    }
+
+    /**
+     * @param array<object> $events
+     */
+    private function dispatchEvents(array $events): void
+    {
+        foreach ($events as $event) {
+            $this->eventDispatcher->dispatch($event);
+        }
+    }
+
+    public function remove(FeatureId $featureId): void
+    {
+        $this->featureRepository->remove($featureId);
```

Comment:
The remove method should dispatch events too

```suggestion
        $this->featureRepository->remove($featureId);
        
        $this->dispatchEvents($feature->release());
```

Reply:
If we need it we should update the `remove` contract because now it expects a `FeatureId` instead of a `Feature` entity. What do you think?

Reply:
Uhmmm, nice point, we have to check what changes but you are right, we need the feature object here

Reply:
I'm going to check it and I might refactor this part!

Reply:
done!